torch/distributed/fsdp/_fully_shard/_fsdp_collectives.py

```diff
@@ -569,6 +569,7 @@ def _get_gradient_divide_factors(
 ) -> Union[tuple[None, None], tuple[float, float]]:
     # For fp32/bf16, we do not need to worry about overflow/underflow, so we
     # use NCCL's built-in division to avoid separate div kernels
+    # Warning: NCCL ReduceOp.AVG may produce incorrect results with world size 1.
```

Comment:
hmm I don't understand why adding a "warning" comment could help resolve the issue mentioned in your summary?

What is this warning used for? Is it only a note for FSDP2 developers who developing the code?